st3/lsp_utils/_client_handler/decorator.py

```diff
@@ -0,0 +1,39 @@
+from LSP.plugin.core.typing import Any, Callable, Iterable, Union
+
+__all__ = [
+    "as_notification_handler",
+    "as_request_handler",
+]
+
+# the first argument is always "self"
+T_HANDLER = Callable[[Any, Any], None]
+T_SERVER_EVENTS = Union[str, Iterable[str]]
+
+HANDLER_MARKS = {
+    "notification": "__handle_notification_events",
+    "request": "__handle_request_events",
+}
+
+
+def as_notification_handler(server_events: T_SERVER_EVENTS) -> Callable[[T_HANDLER], T_HANDLER]:
```

Comment:
I think the `as_` prefix is redundant. A decorator like `@notification_handler` is as clear and probably more conventional (built-in python decorators like `@classmethod` don't use the `as_` prefix).

Reply:
I am fine with whatever names :)